examples/ucp_client_server.c

```diff
@@ -95,6 +96,95 @@ static struct {
  */
 static void usage(void);
 
+void buffer_free(ucp_dt_iov_t *iov)
+{
+    size_t idx;
+
+    for (idx = 0; idx < iov_cnt; idx++) {
+        if (iov[idx].buffer == NULL) {
+            continue;
+        }
```

Comment:
why would it be NULL?

Reply:
@yosefe In ```buffer_malloc```, it use ```iov = calloc(iov_cnt, sizeof(ucp_dt_iov_t));```. If ```buffer_malloc``` failed intermediately, then ```iov[idx].buffer``` could be NULL.

Reply:
If buffer_malloc, we should not get to this function, should have exited (with error) beforehand

Reply:
@yosefe It already return ```NULL``` if it failes in ```buffer_malloc```. However, do we need to release the intermediate already allocated space?

Reply:
buffer_malloc should be self-contained and if it fails it should rollback whatever it allocated

Reply:
@yosefe Got it. I'll remove the related code.